Created routes for editing and deleting a device

giovana-costa opened the pull request:
routes to edit and delete created

giovana-costa committed:
Created routes for editing and deleting a device

joao-fb commented on src/routes.py:
```diff
@@ -111,6 +111,37 @@ def device_create():
+
+    device_data = request.get_json()
+    ID_device = device_data["DeviceID"]
+    new_area = float(device_data["NewArea"])
```
Eu trocaria isso aqui por
```suggestion
    new_area = int(device_data["NewArea"])
```

Os dados de área são armazenados como int.

giovana-costa replied:
Mas a área não poderia ser um valor quebrado?

joao-fb commented on src/routes.py:
```diff
@@ -111,6 +111,37 @@ def device_create():
 
     return jsonify(res), 200
 
+@api.route('device/edit', methods=["PATCH"])
```
Esqueci de colocar na especificação, mas as rotas são protegidas

joao-fb replied:
Adicionar quebra de linha (duas linhas para a função de cima)

joao-fb commented on src/routes.py:
```diff
@@ -111,6 +111,37 @@ def device_create():
+    return jsonify(res), 200
+
+
+@api.route('device/delete', methods=["DELETE"])
```
Proteção de rota

joao-fb commented on src/routes.py:
```diff
@@ -111,6 +111,37 @@ def device_create():
+@api.route('device/delete', methods=["DELETE"])
+def device_delete():
+    device_data = request.get_json()
+    ID_device = device_data["DeviceID"]
```
Precisamos trocar por idList, recebe lista de IDs.

joao-fb commented on src/routes.py:
```diff
@@ -111,6 +111,37 @@ def device_create():
+    device_data = request.get_json()
+    ID_device = device_data["DeviceID"]
+
+    devices_dao.delete_device(ID_device)
```
E aqui precisamos de um for para deletar cada ID. Depois melhoro isso com um método DAO que recebe uma lista de IDs.

joao-fb reviewed (changes requested):
Considerar comentários feitos para realizar as mudanças. Depois só dar commit e push na branch edit/delete

giovana-costa committed:
changing input to Id list

giovana-costa committed:
changing input to Id list

joao-fb commented on src/routes.py:
```diff
@@ -112,6 +112,42 @@ def device_create():
+    """Route for a user to edit a device values."""
+    device_data = request.get_json()
+    ID_device = device_data["DeviceID"]
+    new_area = float(device_data["NewArea"])
```
```suggestion
    new_area = int(device_data["newArea"])
```

Ainda precisa mudar isso aqui, Gi!

joao-fb reviewed (changes requested):
Tudo certo! To aprovando aqui!

joao-fb commented on src/routes.py:
```diff
@@ -112,6 +112,42 @@ def device_create():
+def device_edit():
+    """Route for a user to edit a device values."""
+    device_data = request.get_json()
+    ID_device = device_data["DeviceID"]
```
```suggestion
    ID_device = device_data["deviceID"]
```
```suggestion
    ID_device = device_data["DeviceID"]
```

joao-fb reviewed (changes requested):
Falta pouco hehe

joao-fb committed:
Replacing conversion from float to int

Instead of converting the area to float type, the code will convert it to int. Besides, removed capital letter from the first part of newArea.

joao-fb committed:
Removing capitalized letter

Instead of DeviceID, deviceID
